pkg/controller/assignmetadata/assignmetadata_controller.go

```diff
@@ -119,19 +134,25 @@ func (r *AssignMetadataReconciler) Reconcile(request reconcile.Request) (reconci
 		}
 	}
 	deleted = deleted || !assignMetadata.GetDeletionTimestamp().IsZero()
+	tracker := r.tracker.For(gvkAssignMetadata)
```

Comment:
I've haven't been following closely enough so please excuse the basic questions:
It seems like this controller never retries and always gives up on the first failure, is that correct?

Reply:
Yes, currently it would just try it only once.
We have #989 where we are trying to figure out how to properly handle mutator parsing errors. Some errors are unrecoverable, and in that case we should not retry, but there are also recoverable errors (like conflicts with other Assigns, which can be fixed by altering/removing other Assigns), in which case we should probably wait with `CancelExpect`.